Assets/AI/Actions/ChooseFreePosition.cs

```diff
@@ -0,0 +1,49 @@
+using UnityEngine;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using RAIN.Action;
+using RAIN.Core;
+using RAIN.Representation;
+using RAIN.Navigation;
+
+[RAINAction("Choose Free Position")]
+public class ChooseFreePosition : RAINAction
+{
+
+  public Expression MoveDistance = new Expression();
+
+  public Expression StayOnGraph = new Expression();
+
+  public Expression MoveTargetVariable = new Expression();
+
+  private float _defaultMoveDistance = 10f;
+
+  public override ActionResult Execute(RAIN.Core.AI ai)
+  {
+    if (!MoveTargetVariable.IsVariable)
+      throw new Exception("The Choose Move Position node requires a valid Move Target Variable");
+
+    float tMoveDistance = 0f;
```

Comment:
[ADVICE]
Use underscore in method block
https://github.com/backstreetwx/Pac-Man/wiki/Coding-Standard#member-block